components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -92,6 +93,297 @@
+     * @param operation
+     * @return
+     */
+    private List<String> getScopeOfOperationsFromExtensions(Operation operation) {
+        Map<String, Object> extensions = operation.getVendorExtensions();
+        if (extensions.containsKey(APIConstants.SWAGGER_X_SCOPE)) {
+            String scopeKey = (String) extensions.get(APIConstants.SWAGGER_X_SCOPE);
+            return Collections.singletonList(scopeKey);
+        }
+        return Collections.emptyList();
+    }
+
+    /**
+     * Get parsed Swagger object
+     *
+     * @param oasDefinition OAS definition
+     * @return Swagger
+     * @throws APIManagementException
+     */
+    Swagger getSwagger(String oasDefinition) {
+        SwaggerParser parser = new SwaggerParser();
+        SwaggerDeserializationResult parseAttemptForV2 = parser.readWithInfo(oasDefinition);
+        if (CollectionUtils.isNotEmpty(parseAttemptForV2.getMessages())) {
+            log.debug("Errors found when parsing OAS definition");
+        }
+        return parseAttemptForV2.getSwagger();
+    }
+
+    /**
+     * This method returns the boolean value which checks whether the swagge is included default security scheme or not
+     *
+     * @param swaggerContent resource json
+     * @return is default is given already
+     * @throws APIManagementException
+     */
+    private boolean isDefaultGiven(String swaggerContent) throws APIManagementException {
+        Swagger swagger = getSwagger(swaggerContent);
+        boolean isDefaultIsGiven = true;
+        Map<String, SecuritySchemeDefinition> securityDefinitions = swagger.getSecurityDefinitions();
+        OAuth2Definition checkDefault = (OAuth2Definition) securityDefinitions.get(SWAGGER_SECURITY_SCHEMA_KEY);
+        if (checkDefault == null) {
+            isDefaultIsGiven = false;
+        }
+        return isDefaultIsGiven;
+    }
+
+    /**
+     * This method returns the oauth scopes according to the given swagger(version 2)
+     *
+     * @param swaggerContent resource json
+     * @return scope set as all defaults
+     * @throws APIManagementException
+     */
+    @Override
+    public Set<Scope> injectOtherScopesToDefaultScheme(String swaggerContent) throws APIManagementException {
+        Swagger swagger = getSwagger(swaggerContent);
+
+        Map<String, SecuritySchemeDefinition> securityDefinitions = swagger.getSecurityDefinitions();
+        OAuth2Definition oAuth2Definition;
+        List<String> otherSetOfSchemes = new ArrayList<>();
+        Set<Scope> scopeSet = new HashSet<>();
+        boolean isDefaultAvailable = isDefaultGiven(swaggerContent);
+
+        if (securityDefinitions != null && !isDefaultAvailable) {
+            //If there is no default type schemes set a one
+            OAuth2Definition newDefault = new OAuth2Definition();
+            securityDefinitions.put(SWAGGER_SECURITY_SCHEMA_KEY, newDefault);
+            for (Map.Entry<String, SecuritySchemeDefinition> definition : securityDefinitions.entrySet()) {
```

Comment:
Did we check with a swagger that contains a different security type like api-key along with OAuth non-default scheme?

```
securityDefinitions:
  api_key:
    type: apiKey
    name: api_key
    in: header
  petstore_auth:
    type: oauth2
    authorizationUrl: 'https://petstore.swagger.io/oauth/authorize'
    flow: implicit
    scopes:
      'read:pets': read your pets
      'write:pets': modify pets in your account
```